convert_fendl.py

```diff
@@ -41,90 +42,325 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
-    data = openmc.data.IncidentNeutron.from_ace(filename)
+    if particle == 'neutron':
+        # Get a list of all ACE files, excluding files ending with _ which are 
+        # old incorrect files kept in the release for backwards compatability
+        neutron_files = [
+            f
+            for f in particle_details['ace_files']
+            if not f.name.endswith('_') and not f.name.endswith('.xsd')
+        ]
+        for filename in sorted(neutron_files):
+            # Check for Inf values in K-39 ace file for FENDL-3.0
+            if args.release == '3.0' and filename.name == '19K_039.ace':
+                # Check for the error in case user has provided a fixed version.
+                if 'Inf' in open(filename, 'r').read():
+                    ace_error_warning = """
+                    WARNING: {} contains 'Inf' values within the XSS array which 
+                    prevent conversion to a hdf5 file format. This is a known issue
+                    in FENDL-3.0. {} has not been added to the cross section library 
+                    """.format(filename, filename.name)
+                    ace_error_warning = dedent(ace_error_warning)
+                    warn_k39 = True
+                    continue
+            
+            print(f'Converting: {filename}')
+            data = openmc.data.IncidentNeutron.from_ace(filename)
 
-    # Export HDF5 file
-    h5_file = args.destination / f'{data.name}.h5'
-    print('Writing {}...'.format(h5_file))
-    data.export_to_hdf5(h5_file, 'w', libver=args.libver)
+            # Export HDF5 file
+            h5_file = particle_destination / f'{data.name}.h5'
+            print(f'Writing {h5_file}...')
+            data.export_to_hdf5(h5_file, 'w', libver=args.libver)
 
-    # Register with library
-    library.register_file(h5_file)
+            # Register with library
+            library.register_file(h5_file)
+
+        # Remove the ace files if required    
+        if args.cleanup and ace_files_dir.exists():
+            rmtree(ace_files_dir)
+
+    elif particle == 'photon':
+        for photo_path in sorted(particle_details['photo_files']):
+            print(f'Converting: {photo_path}')
+            data = openmc.data.IncidentPhoton.from_endf(photo_path)
+
+            # Export HDF5 file
+            h5_file = particle_destination / f'{data.name}.h5'
+            print(f'Writing {h5_file}...')
+            data.export_to_hdf5(h5_file, 'w', libver=args.libver)
+
+            # Register with library
+            library.register_file(h5_file)
+        
+        # Remove the ENDF files if required 
+        if args.cleanup and endf_files_dir.exists():
+            rmtree(endf_files_dir)
 
 # Write cross_sections.xml
+print('Writing ', args.destination / 'cross_sections.xml')
 library.export_to_xml(args.destination / 'cross_sections.xml')
+
+# Print the K-39 warning at the end so it doesn't get lost in the conversion messages
+if warn_k39:
+    print(ace_error_warning)
```

Comment:
Use the [`warnings.warn`](https://docs.python.org/3/library/warnings.html#warnings.warn) function from the standard library

Reply:
Is it possible to make warnings raised this way appear at the end of the script? The reason I implemented it like this is so that this warning didn't get lost in all of the writing/converting output lines. 

I saw that this might be possible with `warnings.catch_warnings` but this would mean putting the whole script within the `catch_warnings` context.